src/customJs/properties/promo_codes/PromoCodesWidget.test.js

```diff
@@ -0,0 +1,135 @@
+import { React } from '../../unlayer-react';
+import { render, screen } from '@testing-library/react';
+import { PromoCodesWidget } from './PromoCodesWidget';
+import { EMPTY_SELECTION } from '../../constants';
+import { requestDopplerApp } from '../../utils/dopplerAppBridge';
+
+jest.mock('../../utils/dopplerAppBridge');
+
+// TODO: this is a shared code, make it a common helper
+function prepareUnlayerGlobalObject() {
+  window.unlayer = {
+    setLocale: jest.fn(),
+    setTranslations: jest.fn(),
+  };
+  return window.unlayer;
+}
+
+describe('Promocode widget', () => {
+  it('must be render no options for EMPTY_SELECTION value', async () => {
+    const unlayerPropertyProps = {
+      values: {
+        store: EMPTY_SELECTION,
+      },
+    };
+
+    prepareUnlayerGlobalObject();
+    render(<PromoCodesWidget {...unlayerPropertyProps} />);
+
+    const container = await screen.findByRole('container');
+    expect(container).toBeDefined();
+
+    const radioInput = screen.queryByText('promoCodeRadioInput');
+    expect(radioInput).toBeNull();
+  });
+
+  const unlayerPropertyProps = {
+    value: 'promoCodeTest',
+    values: {
+      store: 'mercadoshops',
+      promo_code: 'promoCodeTest',
+    },
+  };
+
+  it('must be render options by a radio button input for MercadoShop value', async () => {
+    requestDopplerApp.mockImplementation((params) => {
+      params.callback([
+        {
+          code: 'abc-1',
+          type: 'money',
+          value: 1000,
+        },
+      ]);
+      const destructor = () => console.log('destructor mock!');
```

Comment:
What do you think about simplifying this?

```suggestion
      const destructor = noop;
```

or 

```suggestion
      const destructor = () => {};
```